File: agriTech/src/main/java/com/example/agriTech/model/Notification.java
package com.example.agriTech.model;



import java.time.LocalDateTime;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.Data;

@Data
@Entity
@Table(name = "notifications")
public class Notification {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @ManyToOne
    @JoinColumn(name = "device_id")
    private Device device;
    private String title;
    private String message;
    @Column(name = "is_read")
    private Boolean isRead = false;
    @Column(name = "created_at")
    private LocalDateTime createdAt;

}

File: agriTech(DA3K)/app/src/main/java/com/example/agritechda3k/model/Notification.kt
package com.example.agritechda3k.model

import androidx.room.Entity
import androidx.room.PrimaryKey
@Entity(tableName = "notifications")
data class Notification(
    @PrimaryKey val id: Long, // Lấy ID từ Server để không bị trùng
    val userId: Long,
    val plantUserId: Long,
    val title: String,
    val message: String,
    var isRead: Boolean,
    val createdAt: String,
    val deviceCode: String?

)
